### ISSUE TITLE
iquest `not like` operator not working for DATA_RESC_HIER

### ISSUE THREAD

**Issue by tedgin** [irods/irods#6101]
- [x] main
- [x] 4-3-stable
- [x] 4-2-stable
---
I'm using iCommands version 4.2.8.

The `not` portion of the `not like` operator appears to be ignored by `iquest`, meaning that the condition _A not like B_ and _A like B_ yield the same results. Here's an example.

```
ipc_admin@prod ~? iquest '%s' "select DATA_RESC_HIER where DATA_ID = '726482176' and DATA_RESC_HIER not like 'tacc%'"
taccCorralRes;tacc

ipc_admin@prod /iplant? iquest '%s' "select DATA_RESC_HIER where DATA_ID = '726482176' and DATA_RESC_HIER like 'tacc%'"
taccCorralRes;tacc
```

**Comment by alanking** [comment]
It seems that the `banana` operator also functions like `like` and `not like`:
```
$ iquest "select DATA_RESC_HIER where DATA_NAME = 'foo' and DATA_RESC_HIER banana 'demo%'"
DATA_RESC_HIER = demoResc
------------------------------------------------------------
```
The logic for maintaining the facade for querying on `DATA_RESC_HIER` in GenQuery does not consider anything in the condition string other than the presence or absence of "=". In its absence, anything (including empty string) is treated like "like". This includes "!=", which is treated just like "=".

This may fall under the #6168 umbrella (i.e. use a real parser), but it may be worthwhile to fix this for 4.2.x users.

**Comment by trel** [comment]
Yes, I think we should tighten up that condition check for our known 4.2.x tokens.

Open question whether work on #6168 will/does handle this situation already (@korydraughn)...

**Comment by korydraughn** [comment]
Yes. The syntax for the GenQuery rewrite is already protected from things like that due to the flex/bison implementation.

At this time, `DATA_RESC_HIER` is not implemented. That is one of the final items to handle.

**Comment by trel** [comment]
Nice.    So yes, please just make 4.2.x a bit safer/smarter and this can be closed as 4-2-stable only.

**Comment by alanking** [comment]
We'll just include this in each branch for consistency until the new GenQuery parser is ready.

**Comment by tedgin** [comment]
For posterity, this bug also broke iquest queries that use `||` with `DATA_RESC_HIER`.

Here's an example:
```console
rods@dev ~? iquest "select RESC_NAME where DATA_RESC_HIER like 'ingestRes;%'"
RESC_NAME = ingest
------------------------------------------------------------
rods@dev ~? iquest "select RESC_NAME where DATA_RESC_HIER = 'ingestRes' || like 'ingestRes;%'"
CAT_NO_ROWS_FOUND: Nothing was found matching your query
```

**Comment by trel** [comment]
Hi @tedgin ... are you saying

1) the bug existed prior to the commits for this issue?

And also perhaps...

2) that this query again works as expected in this run-up to 4.2.12?

Sounds like we might need another test added to the suite...  if so, we could make a new issue (and link) rather than re-opening this one.

**Comment by tedgin** [comment]
Yep. I'm saying that the bug exists in iRODS 4.2.8, and I assume it was fixed for the upcoming iRODS 4.2.12. I haven't verified that though. I'm an optimist!

**Comment by trel** [comment]
assume != test

:)

https://github.com/irods/irods/issues/6912
